Automattic/vip-go-mu-plugins
# Streamwrapper: Cache file_exists requests

## Description

Start caching `file_exists()`/`stat()` requests to reduce number requests to the Fileservice.

Some actions like image editing can result in multiple `file_exists()` calls which may lead to timeouts on high traffic sites.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Set `VIP_FILESYSTEM_STREAM_WRAPPER_DEBUG` constant as `true` on a test site.
1. Do some media action i.e.: upload file, edit image etc.
1. Check number of `file_exists` requests ( Has `X-action: file_exists` header )
1. Checkout PR and repeat media actions
1. Verify number of `file_exists` requests is reduced.
